common/channel/src/libra_channel.rs

```diff
@@ -52,9 +54,9 @@ impl<T: MessageQueue> Sender<T> {
     /// This adds the message into the internal queue data structure. This is a non-blocking
     /// synchronous call.
     /// TODO: We can have this return a boolean if the queue of a validator is capacity
-    pub fn put(&mut self, message: <T as MessageQueue>::Message) {
+    pub fn put(&mut self, key: <T as MessageQueue>::Key, message: <T as MessageQueue>::Message) {
```

Comment:
You should just be able to write `T::Key` and `T::Message` instead of `<T as XXX>::Type`